src/coreComponents/unitTests/virtualElementTests/testConformingVirtualElementOrder1.cpp

```diff
@@ -176,51 +205,33 @@ static void testCellsInMeshLevel( MeshLevel const & mesh )
-  arrayView2d< real64 const > const faceCenters = faceManager.faceCenter();
-  arrayView2d< real64 const > const faceNormals = faceManager.faceNormal();
-  arrayView1d< real64 const > const faceAreas = faceManager.faceArea();
   arrayView2d< real64 const > cellCenters = cellSubRegion.getElementCenter();
   arrayView1d< real64 const > cellVolumes = cellSubRegion.getElementVolume();
-
   // Loop over cells.
   localIndex const numCells = cellSubRegion.getElementVolume().size();
-  for( localIndex cellIndex = 0; cellIndex < numCells; ++cellIndex )
+
+  using VEM = ConformingVirtualElementOrder1< MAXCELLNODES, MAXFACENODES >;
+  typename VEM::Initialization initialization;
+  FiniteElementBase::initialize< VEM >( nodeManager, edgeManager,
+                                        faceManager, cellSubRegion,
+                                        initialization );
+
+  forAll< parallelDevicePolicy< > >( numCells, [=] GEOSX_HOST_DEVICE
+                                       ( localIndex const cellIndex )
   {
-    using VEM = ConformingVirtualElementOrder1< MAXCELLNODES, MAXFACENODES >;
-    typename VEM::BasisData basisData;
-    real64 const cellCenter[3] { cellCenters( cellIndex, 0 ),
-                                 cellCenters( cellIndex, 1 ),
-                                 cellCenters( cellIndex, 2 ) };
-    VEM::computeProjectors( cellIndex,
-                            nodesCoords,
-                            cellToNodeMap,
-                            elementToFaceMap,
-                            faceToNodeMap,
-                            faceToEdgeMap,
-                            edgeToNodeMap,
-                            faceCenters,
-                            faceNormals,
-                            faceAreas,
-                            cellCenter,
-                            cellVolumes[cellIndex],
-                            basisData
-                            );
+    typename VEM::StackVariables stack;
+    VEM virtualElement;
+    virtualElement.template setup< VEM >( cellIndex, initialization, stack );
 
-    checkIntegralMeanConsistency< VEM >( basisData );
-    checkIntegralMeanDerivativesConsistency< VEM >( basisData );
-    checkStabilizationMatrixConsistency< VEM >( nodesCoords, cellIndex,
-                                                cellToNodeMap, cellCenters,
-                                                basisData );
-    checkSumOfQuadratureWeights< VEM >( cellVolumes[cellIndex],
-                                        basisData );
-  }
+    checkIntegralMeanConsistency< VEM >( virtualElement, stack );
+    checkIntegralMeanDerivativesConsistency< VEM >( virtualElement, stack );
+    checkStabilizationMatrixConsistency< VEM >( nodesCoords, cellIndex, cellToNodeMap,
+                                                cellCenters, virtualElement, stack );
+    checkSumOfQuadratureWeights< VEM >( cellVolumes[cellIndex], stack );
```

Comment:
If these checks fail when CUDA is on, is this unit test actually going to report a failure? I am asking because I noticed that you `ifndef`'ed out all the `EXPECT_TRUE` in these functions. I am not sure, it is just something to double-check.